pkg/build/controller/buildconfig/buildconfig_controller.go

```diff
@@ -174,9 +176,55 @@ func (c *BuildConfigController) buildConfigAdded(obj interface{}) {
 // buildconfig is updated or there is a relist of buildconfigs
 func (c *BuildConfigController) buildConfigUpdated(old, cur interface{}) {
 	bc := cur.(*buildv1.BuildConfig)
+	oldBuildConfig := old.(*buildv1.BuildConfig)
+	// TODO: Remove the counter and event after the image trigger behavior is removed.
+	if c.imageChangeTriggerCleared(oldBuildConfig, bc) {
+		prometheus.MarkImageChangeTriggerCleared(bc.Namespace, bc.Name, true)
+		c.recorder.Event(bc,
+			corev1.EventTypeWarning,
+			"ImageChangeTriggerCleared",
+			"spec.triggers[*].imagechange.lastTriggeredImageID was cleared, which will trigger a build. This behavior is deprecated and will be removed in a future OpenShift release.")
+	} else {
+		prometheus.MarkImageChangeTriggerCleared(bc.Namespace, bc.Name, false)
+	}
 	c.enqueueBuildConfig(bc)
 }
 
+func (c *BuildConfigController) imageChangeTriggerCleared(old, cur *buildv1.BuildConfig) bool {
+	if old == nil || cur == nil {
+		return false
+	}
+	imageLastTriggers := map[corev1.ObjectReference]string{}
+	for _, oldTrigger := range old.Spec.Triggers {
+		if oldTrigger.ImageChange == nil {
+			continue
+		}
+		from := c.getImageChangeTriggerInputReference(old, oldTrigger)
+		imageLastTriggers[*from] = oldTrigger.ImageChange.LastTriggeredImageID
+	}
+	for _, currentTrigger := range cur.Spec.Triggers {
+		if currentTrigger.ImageChange == nil {
+			continue
+		}
+		from := c.getImageChangeTriggerInputReference(cur, currentTrigger)
+		prev, found := imageLastTriggers[*from]
+		if found && len(prev) > 0 && len(currentTrigger.ImageChange.LastTriggeredImageID) == 0 {
+			return true
+		}
```

Comment:
A lot of the logic here was lifted from the ImageChange trigger controller code.